docs/manual-testing-playbook.md

```diff
@@ -0,0 +1,1001 @@
+# Manual testing playbook
+
+This document serves to describe the state of tailscale as packaged in the snap,
+and to facilitate repeatable manual testing.
+
+There are many shell code snippets in this document,
+designed for you to be able to copy and paste into your shell.
+Assuming the following prerequisites are met,
+you should be able to follow the document,
+running the snippets in order,
+to verify everything described here.
+
+## Setup
+
+### Prerequisites
+
+- LXD running on your local machine
+- enough resources on your local machine to launch two VMs
+- an account on https://tailscale.com
+- a local tailscale snap file (optional; you can also install from the snapstore)
+
+### Set up the environment
+
+For help with debugging, it's recommended to `set -x` in the shell,
+so that the commands run are printed before they are executed:
+
+```bash
+set -x
+```
+
+Launch the VMs.
+Note: VMs are used here, because it's more complex to get tailscale working in a container.
+See https://tailscale.com/kb/1130/lxc-unprivileged and https://tailscale.com/kb/1112/userspace-networking for more information.
+
+```bash
+for vm in tailscale-1 tailscale-2; do
+  lxc launch ubuntu:jammy $vm --vm -c limits.cpu=1 -c limits.memory=4GiB
+done
+```
+
+### Install the Tailscale snap
+
+Wait a few seconds for the LXD VM agent to be running,
+then you can install the snap.
+
+If installing the snap from a local file,
+then push and install the snap.
+You will also need to manually connect the interfaces,
+as in the following script:
+
+```bash
+# adjust path as needed
+TAILSCALE_SNAP_PATH=./tailscale_*_amd64.snap
+
+for vm in tailscale-1 tailscale-2; do
+  lxc file push "$TAILSCALE_SNAP_PATH" "$vm/root/"
+  lxc exec $vm -- sh -c 'snap install --dangerous ./tailscale_*.snap'
+  lxc exec $vm -- snap connect tailscale:firewall-control
+  lxc exec $vm -- snap connect tailscale:network-control
+  lxc exec $vm -- snap connect tailscale:sys-devices-virtual-dmi-ids
+  lxc exec $vm -- snap restart tailscale
+  lxc exec $vm -- snap logs tailscale
+  lxc exec $vm -- snap services
+done
+```
+
+Or if installing from the snap store:
+
+```bash
+for vm in tailscale-1 tailscale-2; do
+  lxc exec $vm -- snap install --edge tailscale
+done
+```
+
+### Bring up the tailnet
+
+Now that everything is installed,
+you can use `tailscale up` to bring up the network.
+This should not be affected by snap confinement;
+as in, it should work as in official docs.
+Note that this command will also prompt for authentication;
+when it does, please click the authentication link and confirm it.
```

Comment:
This part is kind of crucial as without logging in, the rest of the test will not be possible. Can we add a warning or something to highlight this?

Reply:
I'm not sure a warning is necessary, because the command will block until logging in has completed. And if you cancel the command, then you haven't completed the step, and it's clear that a step has been missed - as you say, other tests won't work.

Reply:
Resolving so I can merge, since you approved the changes. :)